packages/harmony/src/components/expandable-nav-item/ExpandableNavItem.tsx

```diff
@@ -0,0 +1,114 @@
+import { useMemo, useState } from 'react'
+
+import { useTheme, CSSObject } from '@emotion/react'
+
+import { HarmonyTheme } from '../../foundations/theme'
+import { IconCaretDown, IconCaretRight } from '../../icons'
+import { Flex } from '../layout/Flex'
+import { Text } from '../text'
+
+import type { ExpandableNavItemProps } from './types'
+
+const getStyles = (
+  theme: HarmonyTheme,
+  isOpen: boolean,
+  isHovered: boolean
+): CSSObject => {
+  const baseStyles: CSSObject = {
+    transition: `background-color ${theme.motion.hover}`,
+    cursor: 'pointer',
+    border: `1px solid ${isOpen ? theme.color.border.default : 'transparent'}`
+  }
+
+  const hoverStyles: CSSObject = {
+    backgroundColor: theme.color.background.surface2
+  }
+
+  return {
+    ...baseStyles,
+    ...(isHovered && hoverStyles)
+  }
+}
+
+export const ExpandableNavItem = ({
+  label,
+  leftIcon: LeftIcon,
+  rightIcon,
+  defaultIsOpen = false,
+  nestedItems,
+  shouldPersistRightIcon = false,
+  ...props
+}: ExpandableNavItemProps) => {
+  const [isOpen, setIsOpen] = useState(defaultIsOpen)
+  const [isHovered, setIsHovered] = useState(false)
+  const theme = useTheme()
+
+  const handleMouseEnter = () => setIsHovered(true)
+  const handleMouseLeave = () => setIsHovered(false)
+  const handleClick = () => setIsOpen(!isOpen)
+
+  const styles = useMemo(
+    () => getStyles(theme, isOpen, isHovered),
+    [theme, isOpen, isHovered]
+  )
+
+  const IconComponent = isHovered
+    ? isOpen
+      ? IconCaretDown
+      : IconCaretRight
+    : LeftIcon
+
+  return (
+    <Flex direction='column' {...props}>
+      <Flex
+        alignItems='center'
+        gap='s'
+        pl='s'
+        pr='s'
+        css={{
+          width: '240px',
+          cursor: 'pointer',
+          transition: `background-color ${theme.motion.hover}`
+        }}
+      >
+        <Flex
+          alignItems='center'
+          flex={1}
+          gap='m'
+          p='s'
+          borderRadius='m'
+          css={styles}
+          onMouseEnter={handleMouseEnter}
+          onMouseLeave={handleMouseLeave}
+          onClick={handleClick}
+        >
+          <Flex
+            alignItems='center'
+            gap='m'
+            flex={1}
+            css={{
+              maxWidth: '240px'
+            }}
+          >
+            {IconComponent ? <IconComponent color='default' size='m' /> : null}
+            <Text
+              variant='title'
+              size='l'
+              strength='weak'
+              lineHeight='single'
+              color='default'
+              ellipses
+              maxLines={1}
+            >
+              {label}
+            </Text>
+          </Flex>
+          {isOpen || shouldPersistRightIcon ? rightIcon : null}
+        </Flex>
+      </Flex>
+      {isOpen && nestedItems ? (
```

Comment:
I assume we want some nice animations on expand and close. This can be a follow-up, let's discuss how we want to handle this, likely using react-spring, but im open to other ideas.

Smaller thing, but an fyi, for a11y it's sometimes considered optimal to ensure the collapsed content can still be read by a screen-reader (especially for nav content), which this doesn't do since it's only renders when active. Let's perhaps not overengineer that part right now, but do keep it in the back of your mind as a follow on item to learn about. As a company we don't over-emphasize a11y, but as an engineer i think can be helpful context to implement the best solution.